tests/drum/test_units.py

```diff
@@ -726,6 +766,26 @@ def test_yaml_metadata__multi_hyper_param_optional_fields(
         assert optional_field not in model_metadata[ModelMetadataKeys.HYPERPARAMETERS][0]
 
 
+@pytest.mark.parametrize(
+    "optional_component_param_keys",
+    itertools.combinations(ModelMetadataMultiHyperParamTypes.all(), 1),
+)
+def test_yaml_metadata__multi_hyper_param_optional_component_params(
+    optional_component_param_keys, complete_multi_hyper_param, tmp_path, basic_model_metadata_yaml,
+):
+    model_metadata = basic_model_metadata_yaml
+    for param_key in optional_component_param_keys:
+        complete_multi_hyper_param["values"].pop(param_key)
```

Comment:
What are we iterating here? shouldn't we only pop one at a time to test it works?

Reply:
This is related with the way I pick each of the optional component parameter to test (`itertools.combination(['int', 'float', 'select'], 1)` returns a iterator). This was updated as follows:
https://github.com/datarobot/datarobot-user-models/blob/e6bdbed5a969312cbb201ad0d48d6c804a26beee/tests/drum/test_units.py#L770-L780